src/encoded/static/components/filegallery.js

```diff
@@ -2975,7 +2987,9 @@ class FileGalleryRendererComponent extends React.Component {
         this.experimentType = props.context['@type'][0];
 
         const seriesFiles = getSeriesFiles(props.context) || [];
-        const datasetFiles = [...props.data, ...seriesFiles];
+        const elementsReferenceFiles = props.context.elements_references?.map((elementsReference) => elementsReference.files).reduce((acc, val) => acc.concat(val), []) || [];
```

Comment:
Maybe move this into a utility function shared here and in your first change above? They do the same thing, and a well-named utility function would help documentation.